tools/package-source.js

```diff
@@ -1712,6 +1711,37 @@ _.extend(PackageSource.prototype, {
     return _.keys(packages);
   },
 
+  // Returns an array of objects, representing this package's public
+  // exports. Each object has the following keys:
+  //  - name: export name (ex: "Accounts")
+  //  - arch: an array of strings representing architectures for which this
+  //    export is declared.
+  //
+  // This ignores testOnly exports.
+  getExports: function () {
+    var self = this;
+    var ret = {};
+    // Go over all of the architectures, and aggregate the exports together.
+    _.each(self.architectures, function (arch) {
+      _.each(arch.declaredExports, function (exp) {
+        // Skip testOnly exports -- the flag is intended for use in testing
```

Comment:
What I mean, specifically, is that if packA exports A as testOnly, only it's local test package(localTest:packA, IIRC) should have access to A. Keeps the same interface for most users. 

Not really sure about this.